### DISCUSSION TITLE
Allow hiding directories containing no filter results

### DISCUSSION THREAD

**Discussion by przepompownia** [nvim-tree/nvim-tree.lua#2496] [Ideas]
Hey,

Sometimes I'd like to filter some tree using nvim-tree (usually outside the current project where I am used to Telescope). I didn't found how to hide directories that contains no visible nodes (especially with applied some filter). For example, like in https://github.com/nvim-tree/nvim-tree.lua/issues/2495, after filtering with `wat` query (and optionally hiding hidden files), we see lots of directories that lead to no result.

![image](https://github.com/nvim-tree/nvim-tree.lua/assets/11404453/96817f86-4eba-44f0-92dc-f4f740f06af6)

It would be nice to can hide them using keymap and configuration (at least when the filter query is not empty) and optionally apply `group_empty` on visible ones, if enabled.

**Comment by alex-courtis** [comment]
It sounds like we want the live filter to behave like the others e.g. git clean: hide empty directories and collapse.

Let's follow in #2495 and see how we go.

I'm happy to class this as a bug as the current functionality is not consistent nor useful.
